diesel/src/pg/types/numeric.rs

```diff
@@ -0,0 +1,111 @@
+#[cfg(feature="bigdecimal")]
+mod bigdecimal {
+    extern crate num_traits;
+    extern crate num_bigint;
+    extern crate num_integer;
+    extern crate bigdecimal;
+
+    use std::error::Error;
+    use std::io::prelude::*;
+
+    use pg::Pg;
+
+    use self::num_traits::{Signed, Zero, ToPrimitive};
+    use self::num_bigint::{Sign, BigInt, BigUint};
+    use self::num_integer::Integer;
+    use self::bigdecimal::BigDecimal;
+
+    use pg::data_types::PgNumeric;
+    use types::{self, FromSql, ToSql, IsNull};
+
+    impl ToSql<types::Numeric, Pg> for BigDecimal {
+        fn to_sql<W: Write>(&self, out: &mut W) -> Result<IsNull, Box<Error + Send + Sync>> {
+            use std::str::FromStr;
+
+            let absolute = self.abs();
+
+            let mut digits = vec![];
```

Comment:
This vector's length is always > 0, right? Maybe initialize it with `Vec::with_capacity(n)` where `n` is a "usual" number of digits?